### PR TITLE
Add go-mongox

### PR DESCRIPTION
**Pull request by chenmingyong0423**
## We want to ensure high quality of the packages. Make sure that you've checked the boxes below before sending a pull request.

- [x] I have read the [Contribution Guidelines](https://github.com/avelino/awesome-go/blob/main/CONTRIBUTING.md#contribution-guidelines)
- [x] I have read the [Maintainers Note](https://github.com/avelino/awesome-go/blob/main/CONTRIBUTING.md#maintainers)
- [x] I have read the [Quality Standards](https://github.com/avelino/awesome-go/blob/main/CONTRIBUTING.md#quality-standards)

_Not every repository (project) will require every option, but most projects should. Check the Contribution Guidelines for details._

- [x] The repo documentation has a pkg.go.dev link.
- [x] The repo documentation has a coverage service link.
- [x] The repo documentation has a goreportcard link.
- [x] The repo has a version-numbered release and a go.mod file.
- [x] The repo has a continuous integration process that automatically runs tests that must pass before new pull requests are merged.
- [x] Continuous integration is used to attempt to catch issues prior to releasing this package to end-users.

## Please provide some links to your package to ease the review

- [x] forge link (github.com, gitlab.com, etc): https://github.com/chenmingyong0423/go-mongox
- [x] pkg.go.dev: https://pkg.go.dev/github.com/chenmingyong0423/go-mongox
- [x] goreportcard.com: https://goreportcard.com/report/github.com/chenmingyong0423/go-mongox
- [x] coverage service link ([codecov](https://codecov.io/), [coveralls](https://coveralls.io/), etc.): https://app.codecov.io/gh/chenmingyong0423/go-mongox

## Pull Request content

- [x] The package has been added to the list in alphabetical order.
- [x] The package has an appropriate description with correct grammar.
- [x] As far as I know, the package has not been listed here before.

## Category quality

_Note that new categories can be added only when there are 3 packages or more._

Packages added a long time ago might not meet the current guidelines anymore. It would be very helpful if you could check 3-5 packages above and below your submission to ensure that they also still meet the Quality Standards.

Please delete one of the following lines:

- [x] The packages around my addition still meet the Quality Standards.
- [x] I removed the following packages around my addition: (please give a short reason for each removal)

Thanks for your PR, you're awesome! :sunglasses:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -868,9 +868,9 @@ _Libraries for building and using SQL._
 - [gorethink](https://github.com/dancannon/gorethink) - Go language driver for RethinkDB.
 - [goriak](https://github.com/zegl/goriak) - Go language driver for Riak KV.
 - [Kivik](https://github.com/go-kivik/kivik) - Kivik provides a common Go and GopherJS client library for CouchDB, PouchDB, and similar databases.
-- [mgm](https://github.com/kamva/mgm) - MongoDB model-based ODM for Go (based on official MongoDB driver).
-- [mgo](https://github.com/globalsign/mgo) - (unmaintained) MongoDB driver for the Go language that implements a rich and well tested selection of features under a very simple API following standard Go idioms.
-- [mongo-go-driver](https://github.com/mongodb/mongo-go-driver) - Official MongoDB driver for the Go language.
+- [mgm](https://github.com/kamva/mgm) - DB model-based ODM for Go (based on official DB driver).
+- [mgo](https://github.com/globalsign/mgo) - (unmaintained) DB driver for the Go language that implements a rich and well tested selection of features under a very simple API following standard Go idioms.
+- [-go-driver](https://github.com/mongodb/mongo-go-driver) - Official MongoDB driver for the Go language.
```
Comment (avelino) [inline README.md:873]: why did you remove the word mongo

File: README.md [hunk 2]
```diff
@@ -868,9 +868,9 @@ _Libraries for building and using SQL._
 - [gorethink](https://github.com/dancannon/gorethink) - Go language driver for RethinkDB.
 - [goriak](https://github.com/zegl/goriak) - Go language driver for Riak KV.
 - [Kivik](https://github.com/go-kivik/kivik) - Kivik provides a common Go and GopherJS client library for CouchDB, PouchDB, and similar databases.
-- [mgm](https://github.com/kamva/mgm) - MongoDB model-based ODM for Go (based on official MongoDB driver).
-- [mgo](https://github.com/globalsign/mgo) - (unmaintained) MongoDB driver for the Go language that implements a rich and well tested selection of features under a very simple API following standard Go idioms.
-- [mongo-go-driver](https://github.com/mongodb/mongo-go-driver) - Official MongoDB driver for the Go language.
+- [mgm](https://github.com/kamva/mgm) - DB model-based ODM for Go (based on official DB driver).
```
Comment (avelino) [inline README.md:871]: is mongo model-based, not only db model-based

**Review by avelino — CHANGES_REQUESTED** [review]

**Comment by avelino** [comment]
fixed alphabetical ordem in tests

File: README.md (hunk 1 above)
Comment (chenmingyong0423) [reply README.md:873]: Sorry, I may have misoperated, I restore it

File: README.md (hunk 2 above)
Comment (chenmingyong0423) [reply README.md:871]: Sorry, I may have misoperated, I restore it
